readthedocs/core/middleware.py

```diff
@@ -192,3 +191,20 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Referrer-Policy'] = settings.SECURE_REFERRER_POLICY
         return response
+
+
+class NullCharactersMiddleware:
+    """
+    Block all requests that contains NULL characters (0x00) on their GET attributes.
```

Comment:
Why are we doing this? Should explain a bit more in this docstring. 
```suggestion
    Block all requests that contains NULL characters (0x00) on their GET attributes.
    
    This is because these requests cause issues in our form handling code,
    and lead to 500s showing up in Sentry.
```